interfaces/repo.go

```diff
@@ -477,17 +507,11 @@ func (r *Repository) Interfaces() *Interfaces {
 		for _, plug := range plugs {
 			// Copy part of the data explicitly, leaving out attrs and apps.
 			p := &Plug{
-				Name:      plug.Name,
-				Snap:      plug.Snap,
-				Interface: plug.Interface,
-				Label:     plug.Label,
-			}
-			// Add connection details
-			for slot := range r.plugSlots[plug] {
-				p.Connections = append(p.Connections, SlotRef{
-					Name: slot.Name,
-					Snap: slot.Snap,
-				})
+				Name:        plug.Name,
+				Snap:        plug.Snap,
+				Interface:   plug.Interface,
+				Label:       plug.Label,
+				Connections: plug.Connections[:],
```

Comment:
The intention here seems to be copying data so the internal representation doesn't leak out, so this is a bug. The backing array is being shared. (not Python :-)